### PR TITLE
test(combineLatest): add subscription assertions

### PR DESCRIPTION
**Pull request by staltz**
Add tests for combineLatest (proto and static) that assert the
subscription and unsubscription times of the inner Observables.

### CODE REVIEW

File: spec/observables/combineLatest-spec.js [hunk 1]
```diff
@@ -28,4 +28,397 @@ describe('Observable.combineLatest', function () {
       done();
     });
   });
```
Comment (staltz) [inline spec/observables/combineLatest-spec.js]: The following tests are just copy-pasted from tests on the operator version of combineLatest. What do people think about this? On one hand we need to test whether the static combineLatest works, but on the other hand it's really duplication of code to-be-maintained.

Comment (kwonoj) [inline spec/observables/combineLatest-spec.js]: Personal preference is code to be strictly guarded even there is some of duplication may occurs. Course it'd be better if it's possible to have common fixture setup to reduce duplication..

**Comment by blesh** [comment]
I think it's totally fine, and almost preferable to write tests that cover the same code twice when that code is exposed via two separate points on the API. Things change, and I'd hate for something to change with the static operator, cause breakage, and not be caught because we didn't have tests around it.

**Comment by blesh** [comment]
merged with 7ca7857 thanks, @staltz

**Pull request closed without merging**